kitchen/modules/weights/src/lib.rs

```diff
@@ -0,0 +1,177 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+// Transaction Weight Examples
+// https://crates.parity.io/sr_primitives/weights/index.html
+use support::{ensure, decl_module, decl_storage, dispatch::{Result, WeighData}};
+use runtime_primitives::weights::{ DispatchClass, Weight, ClassifyDispatch, SimpleDispatchInfo };
+
+pub trait Trait: system::Trait {}
+
+decl_storage! {
+	trait Store for Module<T: Trait> as SimpleMap {
+		StoredValue get(stored_value): u32;
+	}
+}
+
+// A "scale" to weigh transactions. This scale can be used with any transactions that take a
+// single argument of type u32. The ultimate weight of the transaction is the / product of the
+// transaction parameter and the field of this struct.
+pub struct Linear(u32);
+
+// The actual weight calculation happens in the
+// impl WeighData block
+impl WeighData<(&u32,)> for Linear {
+	fn weigh_data(&self, (x,): (&u32,)) -> Weight {
+
+		// Use saturation so that an extremely large parameter value
```

Comment:
This is more specific. Overflow still occurs with saturation, but the overflow value is replaced by the max value for the type that questions
```suggestion
		// Saturation handles overflow by using the max value for the type that overflows
```